File: URI/efilogo.py
to_change = [0]*150

def change(c):
	if to_change[ord(c)]:
		if ord(c) > 91:
			return 'f'
		return 'F'
	return c

letters = ['s','j','b','z','p','v','x','S','J','B','Z','P','V','X']
for l in letters:
	to_change[ord(l)] = 1

while True:
	try:
		word = raw_input()
	except EOFError:
		break
	n = len(word)
	
	cont = 0
	f = '*'
	resp = ""
	for i in range(n):
		x = change(word[i])
		if (x == 'f' or x == 'F') and cont == 0:
			f = x
			cont += 1
		elif x == 'f' or x == 'F':
			cont += 1
		else:
			if cont > 0:
				resp += f
				cont = 0
			resp += x
	
	if cont > 0:
		resp += f
	
	print resp

File: To Define/efilogo.cpp
#include <stdio.h>
#include <algorithm>
#include <string.h>

using namespace std;

int to_change[150];

char change(char c) {
    if (to_change[c]) {
        if (c - 'A' > 30) return 'f';
        return 'F';
    }
    return c;
}

int main() {
    char letters[] = {'s','j','b','z','p','v','x','S','J','B','Z','P','V','X'};
    for (char c : letters)
        to_change[c] = 1;

    char word[11000];
    while (fgets(word, sizeof word, stdin)) {
        if (feof(stdin)) break;
        int n = strlen(word);
        for (int i = 0; i < n; i++)
            word[i] = change(word[i]);

        int cont = 0;
        char f;
        for (int i = 0; i < n; i++) {
            if ((word[i] == 'f' || word[i] == 'F') && cont == 0) {
                f = word[i];
                cont++;
            }
            else if (word[i] == 'f' || word[i] == 'F') cont++;
            else {
                if (cont > 0) {
                    printf("%c", f);
                    cont = 0;
                }
                printf("%c", word[i]);
            }
        }

        if (cont > 0) printf("%c", f);
    }
    return 0;
}
